Rover/mode.cpp

```diff
@@ -44,7 +44,10 @@ bool Mode::enter()
 
     // initialisation common to all modes
     if (ret) {
-        set_reversed(false);
+        // reset reversed for autopilot mode except auto
+        if (is_autopilot_mode() && mode_number() != Mode::Number::AUTO) {
```

Comment:
```suggestion
        init_reversed_flag()
```

... and crate a virtual method on the Mode object which checks `is_autopilot_mode`, and overridden by ModeAuto to check the is_resume flag.

Reply:
@peterbarker Thank you for your comment. I'm not sure if I understood your advice properly, but I fixed it.